concent_api/core/tests/test_core_views.py

```diff
@@ -286,6 +286,40 @@ def test_send_reject_message_save_as_receive_out_of_band_status(self):
         self.assertEqual(reject_response.status_code, 202)
         self.assertEqual(Message.objects.last().type, message.RejectReportComputedTask.TYPE)
 
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_reject_message_when_timestamp_too_old(self):
+        response = self.client.post(
+            reverse('core:send'),
+            data = dump(
+                message.Ping(
+                    timestamp=int(dateutil.parser.parse("2017-11-17 09:40:00").timestamp())
+                ),
+                PROVIDER_PRIVATE_KEY,
+                CONCENT_PUBLIC_KEY),
+            content_type                   = 'application/octet-stream',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii')
+        )
+
+        self.assertEqual(response.status_code, 400)
+        self.assertIn('error', response.json().keys())
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_reject_message_when_timestamp_too_far_in_future(self):
```

Comment:
See my remarks for the other test. They mostly apply to this one too.

Reply:
:+1: